src/main/java/cnLabs/reactivedemo/Controller/TeacherController.java

```diff
@@ -0,0 +1,79 @@
+package cnLabs.reactivedemo.Controller;
+
+import cnLabs.reactivedemo.Model.Teacher;
+import cnLabs.reactivedemo.Service.TeacherService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.server.RouterFunction;
+import org.springframework.web.reactive.function.server.ServerResponse;
+import reactor.core.publisher.Mono;
+
+import static org.springframework.web.reactive.function.server.RequestPredicates.*;
+import static org.springframework.web.reactive.function.server.RouterFunctions.route;
+import static org.springframework.web.reactive.function.server.ServerResponse.ok;
+
+@Component
+public class TeacherController {
+
+    @Autowired
+    TeacherService teacherService;
+
+    @Bean
+    public RouterFunction<ServerResponse> getAllTeachers() {
+        return route(GET("/teachers"),
+                req -> ok()
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .body(teacherService.getAllTeachers(), Teacher.class));
+    }
+
+    // endpoint and route in one go
+    @Bean
+    public RouterFunction<ServerResponse> getTeacher() {
+        return route(GET("/teachers/{id}"),
+                req -> ok()
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .body(teacherService.getTeacher(Integer.parseInt(req.pathVariable("id"))), Teacher.class));
+    }
+
+    @Bean
+    public RouterFunction<ServerResponse> addNewTeacher() {
+        return route(POST("/teachers"), req ->
+                req.bodyToMono(Teacher.class)
```

Comment:
Try to be consistent with Code formatting. 
The more consistent you are with code, the easier to maintain

```suggestion
        return route(POST("/teachers"),
                 req -> req.bodyToMono(Teacher.class)
```